atomic_reactor/plugins/pre_inject_yum_repos.py

```diff
@@ -0,0 +1,273 @@
+        self.log.info("yum repo of koji target: '%s'", path)
+        yum_repo.content = render_yum_repo(repo, escape_dollars=False)
+        for platform in self.platforms:
+            self.yum_repos[platform].append(yum_repo)
+
+    def _inject_into_repo_files(self, build_dir: BuildDir):
+        """Inject repo files into a relative directory inside the build context"""
+        host_repos_path = build_dir.path / RELATIVE_REPOS_PATH
+        self.log.info("creating directory for yum repos: %s", host_repos_path)
+        os.mkdir(host_repos_path)
+        allow_repo_dir_in_dockerignore(build_dir.path)
+
+        for repo in self.yum_repos[build_dir.platform]:
+            # Update every repo accordingly in a repofile
+            # input_buf ---- updated ----> updated_buf
+            with StringIO(repo.content.decode()) as input_buf, StringIO() as updated_buf:
+                for line in input_buf:
+                    updated_buf.write(line)
+                    # Apply sslcacert to every repo in a repofile
+                    if line.lstrip().startswith('[') and self._builder_ca_bundle:
+                        updated_buf.write(f'sslcacert=/tmp/{self._ca_bundle_pem}\n')
+
+                yum_repo = YumRepo(repourl=repo.dst_filename,
+                                   content=updated_buf.getvalue(),
+                                   dst_repos_dir=host_repos_path,
+                                   add_hash=False)
+                yum_repo.write_content()
+
+    def _inject_into_dockerfile(self, build_dir: BuildDir):
+        build_dir.dockerfile.add_lines(
+            "ADD %s* %s" % (RELATIVE_REPOS_PATH, YUM_REPOS_DIR),
+            all_stages=True, at_start=True, skip_scratch=True
+        )
+
+        if self._builder_ca_bundle:
+            shutil.copyfile(
+                self._builder_ca_bundle,
+                build_dir.path / self._ca_bundle_pem
+            )
+            build_dir.dockerfile.add_lines(
+                f'ADD {self._ca_bundle_pem} /tmp/{self._ca_bundle_pem}',
+                all_stages=True, at_start=True, skip_scratch=True
+            )
+
+        if not self.workflow.dockerfile_images.base_from_scratch:
+            build_dir.dockerfile.add_lines(*self._cleanup_lines(build_dir.platform))
+
+    def run(self):
+        """
+        run the plugin
+        """
+        if not self.workflow.dockerfile_images:
+            self.log.info("Skipping plugin, from scratch stage(s) can't add repos")
+            return
+
+        if self.include_koji_repo:
+            self.add_koji_repo()
+        else:
+            self.log.info("'include_koji_repo parameter is set to '%s', not including koji repo",
+                          self.include_koji_repo)
+
+        if self.repourls and not is_scratch_build(self.workflow):
+            self.validate_yum_repo_files_url()
+
+        for platform in self.platforms:
+            fetched_repourls = set()
+            for repourl in self.repourls.get(platform, []):
```

Comment:
and what about 'noarch' arch which is for all arches??? :)

Reply:
it is merged in resolve_composes with all arches.